editor/writing-flow/index.js

```diff
@@ -37,31 +43,63 @@ class WritingFlow extends Component {
 			) );
 	}
 
-	moveFocusInContainer( target, direction = 'UP' ) {
+	moveFocusInContainer( target, reverse, horizontal ) {
 		const focusableNodes = this.getVisibleTabbables();
-		if ( direction === 'UP' ) {
+
+		if ( reverse ) {
 			focusableNodes.reverse();
 		}
 
 		const targetNode = focusableNodes
 			.slice( focusableNodes.indexOf( target ) )
-			.reduce( ( result, node ) => {
-				return result || ( node.contains( target ) ? null : node );
+			.reduce( ( result, node, i, array ) => {
+				if ( result ) {
+					return result;
+				}
+
+				if ( node.contains( target ) ) {
+					return null;
+				}
+
+				const nextNode = array[ i + 1 ];
+
+				// Skip node if it contains a focusable node.
+				if ( nextNode && node.contains( nextNode ) ) {
+					return null;
+				}
+
+				return node;
 			}, null );
 
 		if ( targetNode ) {
-			placeCaretAtEdge( targetNode, direction === 'DOWN' );
+			if ( horizontal ) {
+				placeCaretAtHorizontalEdge( targetNode, reverse );
+			} else {
+				placeCaretAtVerticalEdge( targetNode, reverse );
+			}
 		}
 	}
 
 	onKeyDown( event ) {
 		const { keyCode, target } = event;
-		const moveUp = ( keyCode === UP || keyCode === LEFT );
-		const moveDown = ( keyCode === DOWN || keyCode === RIGHT );
+		const up = keyCode === UP;
+		const down = keyCode === DOWN;
+		const left = keyCode === LEFT;
+		const right = keyCode === RIGHT;
+		const reverse = up || left;
+		const horizontal = left || right;
+		const vertical = up || down;
```

Comment:
As some of the functions here and in `dom/utils` grow, keeping track of boolean variables becomes harder. Aside from a `[modal]+[main verb]` nomenclature (with _modal_ taking values in _{  is, should, can, may, will, … }_),  I'd love to see more descriptive names throughout. Examples:

- `up` → `isUpKey`
- `reverse` → `isAscending`
- `noScroll` → logic reversal, becoming `mayUseScroll`